sklearn/tests/test_calibration.py

```diff
@@ -182,6 +188,68 @@ def test_sample_weight(data, method, ensemble):
     assert diff > 0.1
 
 
+@pytest.mark.parametrize("method", ["sigmoid", "isotonic"])
+@pytest.mark.parametrize("ensemble", [True, False])
+def test_sample_weight_class_imbalanced(method, ensemble):
+    """Use an imbalanced dataset to check that `sample_weight` is taken into
+    account in the calibration estimator."""
+    X, y = make_blobs((100, 1000), center_box=(-1, 1), random_state=42)
+
+    # Compute weights to compensate for the unbalance of the dataset
+    weights = np.array([0.9, 0.1])
+    sample_weight = weights[(y == 1).astype(int)]
+
+    X_train, X_test, y_train, y_test, sw_train, sw_test = train_test_split(
+        X, y, sample_weight, stratify=y, random_state=42
+    )
+
+    # FIXME: ideally we should create a `Pipeline` with the `StandardScaler`
+    # followed by the `LinearSVC`. However, `Pipeline` does not expose
+    # `sample_weight` and it will be silently ignored.
+    scaler = StandardScaler()
+    X_train = scaler.fit_transform(X_train)
+    X_test = scaler.transform(X_test)
+
+    base_estimator = LinearSVC(random_state=42)
+    calibrated_clf = CalibratedClassifierCV(
+        base_estimator, method=method, ensemble=ensemble
+    )
+    calibrated_clf.fit(X_train, y_train, sample_weight=sw_train)
+    predictions = calibrated_clf.predict_proba(X_test)[:, 1]
+
+    assert brier_score_loss(y_test, predictions, sample_weight=sw_test) < 0.2
+
+
+@pytest.mark.parametrize("method", ["sigmoid", "isotonic"])
+def test_sample_weight_class_imbalanced_ensemble_equivalent(method):
+    X, y = make_blobs((100, 1000), center_box=(-1, 1), random_state=42)
+
+    # Compute weigths to compensate the unbalance of the dataset
+    sample_weight = 9 * (y == 0) + 1
+
+    X_train, X_test, y_train, y_test, sw_train, sw_test = train_test_split(
+        X, y, sample_weight, stratify=y, random_state=42
+    )
+
+    scaler = StandardScaler()
+    X_train = scaler.fit_transform(
+        X_train
+    )  # compute mean, std and transform training data as well
+    X_test = scaler.transform(X_test)
```

Comment:
```suggestion
    # FIXME: ideally we should create a `Pipeline` with the `StandardScaler`
    # followed by the `LinearSVC`. However, `Pipeline` does not expose
    # `sample_weight` and it will be silently ignored.
    scaler = StandardScaler()
    X_train = scaler.fit_transform(X_train)
    X_test = scaler.transform(X_test)
```